packages/flutter/lib/src/painting/edge_insets.dart

```diff
@@ -56,6 +65,18 @@ class EdgeInsets {
   /// The total offset in the horizontal direction.
   double get vertical => top + bottom;
 
+  /// The total offset in the given direction.
+  double along(Axis axis) {
```

Comment:
This is great. There are probably a bunch of classes that would benefit from along(Axis) methods. Maybe Size, Offset, FractionalOffset. No need to do them all right away; this seems like a good pattern to apply in the future.